Docker demo fails

Following the Docker setup instructions - docker build went fine, but then the first example fails. 

It looks to me like there is a configuration error in the SDF setup - I am not a ROS/Gazebo expert so I apologize if I'm missing something obvious.
This is on Windows 11 / WSL2 (Ubuntu 24.04)

Any help would be appreciated!

After docker compose run overlay bash:

`root@594663-MITLL:/overlay_ws# ros2 launch tb_worlds tb_demo_world.launch.py
[INFO] [launch]: All log files can be found below /root/.ros/log/2025-07-12-17-34-31-336286-594663-MITLL-968
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [xacro-1]: process started with pid [986]
[INFO] [parameter_bridge-2]: process started with pid [987]
[INFO] [create-3]: process started with pid [988]
[INFO] [gz-4]: process started with pid [989]
[INFO] [robot_state_publisher-5]: process started with pid [990]
[INFO] [rviz2-6]: process started with pid [991]
[INFO] [component_container_isolated-7]: process started with pid [992]
[INFO] [create-8]: process started with pid [993]
[INFO] [create-9]: process started with pid [994]
[INFO] [create-10]: process started with pid [995]
[component_container_isolated-7] [INFO] [1752341672.064753844] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libmap_server_core.so
[robot_state_publisher-5] [INFO] [1752341672.084526394] [robot_state_publisher]: Robot initialized
[create-3] [INFO] [1752341672.095502080] [ros_gz_sim]: Requesting list of world names.
[create-8] [INFO] [1752341672.095534517] [ros_gz_sim]: Requesting list of world names.
[create-9] [INFO] [1752341672.095534741] [ros_gz_sim]: Requesting list of world names.
[create-10] [INFO] [1752341672.095557645] [ros_gz_sim]: Requesting list of world names.
[component_container_isolated-7] [INFO] [1752341672.108370524] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_map_server::CostmapFilterInfoServer>
[component_container_isolated-7] [INFO] [1752341672.108410828] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_map_server::MapSaver>
[component_container_isolated-7] [INFO] [1752341672.108415477] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_map_server::MapServer>
[component_container_isolated-7] [INFO] [1752341672.108418769] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_map_server::MapServer>
[component_container_isolated-7] [INFO] [1752341672.115278575] [map_server]:
[component_container_isolated-7]        map_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.115325928] [map_server]: Creating
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/map_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.115991047] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libamcl_core.so
[component_container_isolated-7] [INFO] [1752341672.124355930] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_amcl::AmclNode>
[component_container_isolated-7] [INFO] [1752341672.124401037] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_amcl::AmclNode>
[INFO] [xacro-1]: process has finished cleanly [pid 986]
[component_container_isolated-7] [INFO] [1752341672.128401254] [amcl]:
[component_container_isolated-7]        amcl lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.129653550] [amcl]: Creating
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/amcl' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.132900512] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libnav2_lifecycle_manager_core.so
[component_container_isolated-7] [INFO] [1752341672.136674505] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_lifecycle_manager::LifecycleManager>
[component_container_isolated-7] [INFO] [1752341672.136706780] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_lifecycle_manager::LifecycleManager>
[component_container_isolated-7] [INFO] [1752341672.142301191] [lifecycle_manager_localization]: Creating
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/lifecycle_manager_localization' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.150706081] [lifecycle_manager_localization]: Creating and initializing lifecycle service clients
[component_container_isolated-7] [INFO] [1752341672.151866232] [lifecycle_manager_localization]: Starting managed nodes bringup...
[component_container_isolated-7] [INFO] [1752341672.151902327] [lifecycle_manager_localization]: Configuring map_server
[component_container_isolated-7] [INFO] [1752341672.152105612] [map_server]: Configuring
[component_container_isolated-7] [INFO] [1752341672.152177902] [map_io]: Loading yaml file: /overlay_ws/install/tb_worlds/share/tb_worlds/maps/sim_house_map.yaml
[component_container_isolated-7] [INFO] [1752341672.152419801] [map_io]: resolution: 0.05
[component_container_isolated-7] [INFO] [1752341672.152426777] [map_io]: origin[0]: -2.07
[component_container_isolated-7] [INFO] [1752341672.152429016] [map_io]: origin[1]: -1.64
[component_container_isolated-7] [INFO] [1752341672.152431226] [map_io]: origin[2]: 0
[component_container_isolated-7] [INFO] [1752341672.152433573] [map_io]: free_thresh: 0.25
[component_container_isolated-7] [INFO] [1752341672.152435613] [map_io]: occupied_thresh: 0.65
[component_container_isolated-7] [INFO] [1752341672.152439440] [map_io]: mode: trinary
[component_container_isolated-7] [INFO] [1752341672.152442127] [map_io]: negate: 0
[component_container_isolated-7] [INFO] [1752341672.155819293] [map_io]: Loading image_file: /overlay_ws/install/tb_worlds/share/tb_worlds/maps/sim_house_map.pgm
[component_container_isolated-7] [INFO] [1752341672.157310046] [map_io]: Read map /overlay_ws/install/tb_worlds/share/tb_worlds/maps/sim_house_map.pgm: 170 X 138 map @ 0.05 m/cell
[component_container_isolated-7] [INFO] [1752341672.158622234] [lifecycle_manager_localization]: Configuring amcl
[component_container_isolated-7] [INFO] [1752341672.158791065] [amcl]: Configuring
[component_container_isolated-7] [INFO] [1752341672.158883142] [amcl]: initTransforms
[component_container_isolated-7] [INFO] [1752341672.163261950] [amcl]: initPubSub
[component_container_isolated-7] [INFO] [1752341672.164041160] [amcl]: Subscribed to map topic.
[component_container_isolated-7] [INFO] [1752341672.166096076] [lifecycle_manager_localization]: Activating map_server
[component_container_isolated-7] [INFO] [1752341672.166241659] [map_server]: Activating
[component_container_isolated-7] [INFO] [1752341672.166303323] [map_server]: Creating bond (map_server) to lifecycle manager.
[component_container_isolated-7] [INFO] [1752341672.166355196] [amcl]: Received a 170 X 138 map @ 0.050 m/pix
[component_container_isolated-7] [INFO] [1752341672.268485040] [lifecycle_manager_localization]: Server map_server connected with bond.
[component_container_isolated-7] [INFO] [1752341672.268545283] [lifecycle_manager_localization]: Activating amcl
[component_container_isolated-7] [INFO] [1752341672.268718801] [amcl]: Activating
[component_container_isolated-7] [INFO] [1752341672.268795231] [amcl]: initialPoseReceived
[component_container_isolated-7] [INFO] [1752341672.268910447] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libcontroller_server_core.so
[component_container_isolated-7] [INFO] [1752341672.269253438] [amcl]: Setting pose (0.000000): 0.600 0.600 0.000
[component_container_isolated-7] [INFO] [1752341672.269600881] [amcl]: Creating bond (amcl) to lifecycle manager.
[component_container_isolated-7] [INFO] [1752341672.281775471] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_controller::ControllerServer>
[component_container_isolated-7] [INFO] [1752341672.281814838] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_controller::ControllerServer>
[component_container_isolated-7] [INFO] [1752341672.286347220] [controller_server]:
[component_container_isolated-7]        controller_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.299281932] [controller_server]: Creating controller server
[component_container_isolated-7] [INFO] [1752341672.304804156] [local_costmap.local_costmap]:
[component_container_isolated-7]        local_costmap lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.306654389] [local_costmap.local_costmap]: Creating Costmap
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/controller_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.310891687] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libsmoother_server_core.so
[component_container_isolated-7] [INFO] [1752341672.315610769] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_smoother::SmootherServer>
[component_container_isolated-7] [INFO] [1752341672.315642948] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_smoother::SmootherServer>
[component_container_isolated-7] [INFO] [1752341672.319706723] [smoother_server]:
[component_container_isolated-7]        smoother_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.320633362] [smoother_server]: Creating smoother server
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/smoother_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.322819846] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libplanner_server_core.so
[component_container_isolated-7] [INFO] [1752341672.325444782] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_planner::PlannerServer>
[component_container_isolated-7] [INFO] [1752341672.325476961] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_planner::PlannerServer>
[component_container_isolated-7] [INFO] [1752341672.329669531] [planner_server]:
[component_container_isolated-7]        planner_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.330644716] [planner_server]: Creating
[component_container_isolated-7] [INFO] [1752341672.335839788] [global_costmap.global_costmap]:
[component_container_isolated-7]        global_costmap lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.336146298] [global_costmap.global_costmap]: Creating Costmap
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/planner_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.339589711] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libbehavior_server_core.so
[component_container_isolated-7] [INFO] [1752341672.341480228] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<behavior_server::BehaviorServer>
[component_container_isolated-7] [INFO] [1752341672.341509999] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<behavior_server::BehaviorServer>
[component_container_isolated-7] [INFO] [1752341672.346065113] [behavior_server]:
[component_container_isolated-7]        behavior_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/behavior_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.350403880] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libbt_navigator_core.so
[component_container_isolated-7] [INFO] [1752341672.353060834] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_bt_navigator::BtNavigator>
[component_container_isolated-7] [INFO] [1752341672.353092073] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_bt_navigator::BtNavigator>
[component_container_isolated-7] [INFO] [1752341672.358635949] [bt_navigator]:
[component_container_isolated-7]        bt_navigator lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.359595591] [bt_navigator]: Creating
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/bt_navigator' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.361524469] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libwaypoint_follower_core.so
[component_container_isolated-7] [INFO] [1752341672.365575759] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_waypoint_follower::WaypointFollower>
[component_container_isolated-7] [INFO] [1752341672.365608135] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_waypoint_follower::WaypointFollower>
[component_container_isolated-7] [INFO] [1752341672.369835295] [waypoint_follower]:
[component_container_isolated-7]        waypoint_follower lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.371481780] [waypoint_follower]: Creating
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/waypoint_follower' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.373737776] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libvelocity_smoother_core.so
[component_container_isolated-7] [INFO] [1752341672.376172573] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_velocity_smoother::VelocitySmoother>
[component_container_isolated-7] [INFO] [1752341672.376354189] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_velocity_smoother::VelocitySmoother>
[component_container_isolated-7] [INFO] [1752341672.381509563] [velocity_smoother]:
[component_container_isolated-7]        velocity_smoother lifecycle node launched.
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/velocity_smoother' in container '/nav2_container'
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.383286492] [lifecycle_manager_localization]: Server amcl connected with bond.
[component_container_isolated-7] [INFO] [1752341672.383318145] [lifecycle_manager_localization]: Managed nodes are active
[component_container_isolated-7] [INFO] [1752341672.383329513] [lifecycle_manager_localization]: Creating bond timer...
[component_container_isolated-7] [INFO] [1752341672.383886867] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libcollision_monitor_core.so
[component_container_isolated-7] [INFO] [1752341672.391738703] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_collision_monitor::CollisionMonitor>
[component_container_isolated-7] [INFO] [1752341672.391771871] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_collision_monitor::CollisionMonitor>
[component_container_isolated-7] [INFO] [1752341672.398608425] [collision_monitor]:
[component_container_isolated-7]        collision_monitor lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/collision_monitor' in container '/nav2_container'
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.401360207] [nav2_container]: Load Library: /opt/ros/jazzy/lib/libopennav_docking_core.so
[component_container_isolated-7] [INFO] [1752341672.406195901] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<opennav_docking::DockingServer>
[component_container_isolated-7] [INFO] [1752341672.406232218] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<opennav_docking::DockingServer>
[component_container_isolated-7] [INFO] [1752341672.411627064] [docking_server]:
[component_container_isolated-7]        docking_server lifecycle node launched.
[component_container_isolated-7]        Waiting on external lifecycle transitions to activate
[component_container_isolated-7]        See https://design.ros2.org/articles/node_lifecycle.html for more information.
[component_container_isolated-7] [INFO] [1752341672.411673571] [docking_server]: Creating docking_server
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/docking_server' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.414223308] [nav2_container]: Found class: rclcpp_components::NodeFactoryTemplate<nav2_lifecycle_manager::LifecycleManager>
[component_container_isolated-7] [INFO] [1752341672.414249806] [nav2_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<nav2_lifecycle_manager::LifecycleManager>
[component_container_isolated-7] [INFO] [1752341672.418478977] [lifecycle_manager_navigation]: Creating
[component_container_isolated-7] [INFO] [1752341672.419520092] [lifecycle_manager_navigation]: Creating and initializing lifecycle service clients
[INFO] [launch_ros.actions.load_composable_nodes]: Loaded node '/lifecycle_manager_navigation' in container '/nav2_container'
[component_container_isolated-7] [INFO] [1752341672.423628435] [lifecycle_manager_navigation]: Starting managed nodes bringup...
[component_container_isolated-7] [INFO] [1752341672.423682501] [lifecycle_manager_navigation]: Configuring controller_server
[component_container_isolated-7] [INFO] [1752341672.423827853] [controller_server]: Configuring controller interface
[component_container_isolated-7] [INFO] [1752341672.423851170] [controller_server]: getting progress checker plugins..
[component_container_isolated-7] [INFO] [1752341672.424036114] [controller_server]: getting goal checker plugins..
[component_container_isolated-7] [INFO] [1752341672.424204484] [controller_server]: Controller frequency set to 20.0000Hz
[component_container_isolated-7] [INFO] [1752341672.424259722] [local_costmap.local_costmap]: Configuring
[component_container_isolated-7] [INFO] [1752341672.427236634] [local_costmap.local_costmap]: Using plugin "voxel_layer"
[component_container_isolated-7] [INFO] [1752341672.436423578] [local_costmap.local_costmap]: Subscribed to Topics: scan
[component_container_isolated-7] [INFO] [1752341672.440360496] [local_costmap.local_costmap]: Initialized plugin "voxel_layer"
[component_container_isolated-7] [INFO] [1752341672.440383750] [local_costmap.local_costmap]: Using plugin "inflation_layer"
[component_container_isolated-7] [INFO] [1752341672.440932452] [local_costmap.local_costmap]: Initialized plugin "inflation_layer"
[component_container_isolated-7] [INFO] [1752341672.447515736] [controller_server]: Created progress_checker : progress_checker of type nav2_controller::SimpleProgressChecker
[component_container_isolated-7] [INFO] [1752341672.447818465] [controller_server]: Controller Server has progress_checker  progress checkers available.
[component_container_isolated-7] [INFO] [1752341672.448857173] [controller_server]: Created goal checker : general_goal_checker of type nav2_controller::SimpleGoalChecker
[component_container_isolated-7] [INFO] [1752341672.449188782] [controller_server]: Controller Server has general_goal_checker  goal checkers available.
[component_container_isolated-7] [INFO] [1752341672.452192274] [controller_server]: Created controller : FollowPath of type nav2_mppi_controller::MPPIController
[component_container_isolated-7] [INFO] [1752341672.454898372] [controller_server]: Controller period is equal to model dt. Control sequence shifting is ON
[component_container_isolated-7] [INFO] [1752341672.458583741] [controller_server]: ConstraintCritic instantiated with 1 power and 4.000000 weight.
[component_container_isolated-7] [INFO] [1752341672.458616742] [controller_server]: Critic loaded : mppi::critics::ConstraintCritic
[component_container_isolated-7] [WARN] [1752341672.459688625] [controller_server]: Inconsistent configuration in collision checking. Please verify the robot's shape settings in both the costmap and the cost critic.
[component_container_isolated-7] Magick: abort due to signal 11 (SIGSEGV) "Segmentation Fault"...
[rviz2-6] [INFO] [1752341672.629145647] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-6] [INFO] [1752341672.629254652] [rviz2]: OpenGl version: 4.5 (GLSL 4.5)
[rviz2-6] [INFO] [1752341672.692196454] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-6] [INFO] [1752341672.803130661] [rviz]: Subscribing to: /mobile_base/sensors/bumper_pointcloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.803610377] [rviz]: Subscribing to: /mobile_base/sensors/bumper_pointcloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.817530754] [rviz]: Subscribing to: /global_costmap/voxel_marked_cloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.822889095] [rviz]: Subscribing to: /local_costmap/voxel_marked_cloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.844301645] [rviz]: Subscribing to: /mobile_base/sensors/bumper_pointcloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.846951748] [rviz]: Subscribing to: /global_costmap/voxel_marked_cloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.848899905] [rviz]: Subscribing to: /local_costmap/voxel_marked_cloud
[rviz2-6]
[rviz2-6] [INFO] [1752341672.883622519] [nav2_rviz_selector_node]: Trying to load plugins...
[parameter_bridge-2] [INFO] [1752341673.114318725] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/clock (gz.msgs.Clock) -> /clock (rosgraph_msgs/msg/Clock)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.117593519] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/joint_states (gz.msgs.Model) -> joint_states (sensor_msgs/msg/JointState)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.120862049] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/odom (gz.msgs.Odometry) -> odom (nav_msgs/msg/Odometry)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.122515715] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/tf (gz.msgs.Pose_V) -> tf (tf2_msgs/msg/TFMessage)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.125139900] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/imu (gz.msgs.IMU) -> imu (sensor_msgs/msg/Imu)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.125614325] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/scan (gz.msgs.LaserScan) -> scan (sensor_msgs/msg/LaserScan)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.126117629] [ros_gz_bridge]: Creating ROS->GZ Bridge: [cmd_vel (geometry_msgs/msg/TwistStamped) -> /cmd_vel (gz.msgs.Twist)] (Lazy 0)
[parameter_bridge-2] [INFO] [1752341673.128558975] [ros_gz_bridge]: Creating GZ->ROS Bridge: [/realsense_camera (gz.msgs.Image) -> /camera/image_raw (sensor_msgs/msg/Image)] (Lazy 0)
[create-9] [INFO] [1752341673.283701734] [ros_gz_sim]: Entity creation successful.
[create-8] [INFO] [1752341673.283708908] [ros_gz_sim]: Entity creation successful.
[create-10] [INFO] [1752341673.283718352] [ros_gz_sim]: Entity creation successful.
[create-3] [INFO] [1752341673.283949270] [ros_gz_sim]: Entity creation successful.
[gz-4] libEGL warning: MESA-LOADER: egl: failed to open vgem: driver not built!
[gz-4]
[gz-4] libEGL warning: NEEDS EXTENSION: falling back to kms_swrast
[INFO] [create-9]: process has finished cleanly [pid 994]
[INFO] [create-3]: process has finished cleanly [pid 988]
[INFO] [create-8]: process has finished cleanly [pid 993]
[INFO] [create-10]: process has finished cleanly [pid 995]
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="imu_link"]/sensor[@name="tb3_imu"]/gz_frame_id:<data-string>:L51]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="base_scan"]/sensor[@name="hls_lfcd_lds"]/gz_frame_id:<data-string>:L138]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="camera_link"]/sensor[@name="intel_realsense_r200_depth"]/gz_frame_id:<data-string>:L368]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] libEGL warning: MESA-LOADER: egl: failed to open vgem: driver not built!
[gz-4]
[gz-4] libEGL warning: NEEDS EXTENSION: falling back to kms_swrast
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="imu_link"]/sensor[@name="tb3_imu"]/gz_frame_id:<data-string>:L94]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="base_scan"]/sensor[@name="hls_lfcd_lds"]/gz_frame_id:<data-string>:L166]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="camera_link"]/sensor[@name="intel_realsense_r200_depth"]/gz_frame_id:<data-string>:L406]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="imu_link"]/sensor[@name="tb3_imu"]/gz_frame_id:<data-string>:L94]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="base_scan"]/sensor[@name="hls_lfcd_lds"]/gz_frame_id:<data-string>:L166]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[gz-4] Warning [Utils.cc:132] [/sdf/model[@name="turtlebot3_waffle"]/link[@name="camera_link"]/sensor[@name="intel_realsense_r200_depth"]/gz_frame_id:<data-string>:L406]: XML Element[gz_frame_id], child of element[sensor], not defined in SDF. Copying[gz_frame_id] as children of [sensor].
[ERROR] [component_container_isolated-7]: process has died [pid 992, exit code -6, cmd '/opt/ros/jazzy/lib/rclcpp_components/component_container_isolated --ros-args --log-level info --ros-args -r __node:=nav2_container --params-file /tmp/launch_params_f5hfou48 --params-file /tmp/launch_params_ng7dtk0_ -r /tf:=tf -r /tf_static:=tf_static'].
[rviz2-6] [INFO] [1752341677.289253723] [rviz]: Message Filter dropping message: frame 'base_scan' at time 0.006 for reason 'discarding message because the queue is full'
[rviz2-6] [INFO] [1752341677.513203366] [rviz]: Message Filter dropping message: frame 'base_scan' at time 0.201 for reason 'discarding message because the queue is full'
[rviz2-6] [INFO] [1752341677.705328489] [rviz]: Message Filter dropping message: frame 'base_scan' at time 0.402 for reason 'discarding message because the queue is full'
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT) again, ignoring...
[parameter_bridge-2] [INFO] [1752341677.890421462] [rclcpp]: signal_handler(signum=2)
[INFO] [parameter_bridge-2]: process has finished cleanly [pid 987]
[INFO] [gz-4]: process has finished cleanly [pid 989]
[INFO] [robot_state_publisher-5]: process has finished cleanly [pid 990]
[robot_state_publisher-5] [INFO] [1752341677.890420897] [rclcpp]: signal_handler(signum=2)
^C[rviz2-6] [INFO] [1752341677.890421377] [rclcpp]: signal_handler(signum=2)
[WARNING] [launch]: user interrupted with ctrl-c (SIGINT) again, ignoring...
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT) again, ignoring...
[ERROR] [rviz2-6]: process has died [pid 991, exit code -11, cmd '/opt/ros/jazzy/lib/rviz2/rviz2 -d /opt/ros/jazzy/share/nav2_bringup/rviz/nav2_default_view.rviz --ros-args --params-file /tmp/launch_params_dx63z_dn'].
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Cannot shutdown a ROS adapter that is not running
root@594663-MITLL:/overlay_ws# grep -rn waffle *
build/tb_worlds/symlink_install_manifest.txt:17:/overlay_ws/install/tb_worlds/share/tb_worlds/urdf/gz_waffl.sdf.xacro
grep: install/tb_worlds/share/tb_worlds/launch/__pycache__/turtlebot_spawner.launch.cpython-312.pyc: binary file matches
grep: install/tb_worlds/share/tb_worlds/launch/__pycache__/tb_world.launch.cpython-312.pyc: binary file matches
log/build_2025-07-12_16-55-08/tb_worlds/streams.log:50:[1.152s] -- Symlinking: /overlay_ws/install/tb_worlds/share/tb_worlds/urdf/gz_waffle.sdf.xacro
log/build_2025-07-12_16-55-08/tb_worlds/stdout_stderr.log:45:-- Symlinking: /overlay_ws/install/tb_worlds/share/tb_worlds/urdf/gz_waffle.sdf.xacro
log/build_2025-07-12_16-55-08/tb_worlds/stdout.log:45:-- Symlinking: /overlay_ws/install/tb_worlds/share/tb_worlds/urdf/gz_waffle.sdf.xacro
log/build_2025-07-12_16-55-08/events.log:68:[1.152228] (tb_worlds) StdoutLine: {'line': b'-- Symlinking: /overlay_ws/install/tb_worlds/share/tb_worlds/urdf/gz_waffle.sdf.xacro\n'}
src/tb_worlds/urdf/gz_waffle.sdf.xacro:5:# https://github.com/ros-navigation/nav2_minimal_turtlebot_simulation/blob/091b5ff12436890a54de6325df3573ae110e912b/nav2_minimal_tb3_sim/urdf/gz_waffle.sdf.xacro
src/tb_worlds/urdf/gz_waffle.sdf.xacro:11:    <model name="turtlebot3_waffle">
src/tb_worlds/urdf/gz_waffle.sdf.xacro:44:              <uri>package://nav2_minimal_tb3_sim/models/turtlebot3_model/meshes/waffle_base.dae</uri>
src/tb_worlds/launch/tb_world.launch.py:73:        default_value=os.path.join(bringup_dir, "urdf", "gz_waffle.sdf.xacro"),
src/tb_worlds/launch/tb_world.launch.py:86:            "turtlebot3_waffle.urdf",
src/tb_worlds/launch/turtlebot_spawner.launch.py:57:        default_value=os.path.join(bringup_dir, "urdf", "gz_waffle.sdf.xacro"),
src/tb_worlds/configs/turtlebot3_bridge.yaml:2:# https://github.com/ros-navigation/nav2_minimal_turtlebot_simulation/blob/091b5ff12436890a54de6325df3573ae110e912b/nav2_minimal_tb3_sim/configs/turtlebot3_waffle_bridge.yaml`

--- a/tb_worlds/configs/nav2_params.yaml
+++ b/tb_worlds/configs/nav2_params.yaml
@@ -1,5 +1,5 @@
 # This file has been modified from
-# https://github.com/ros-navigation/navigation2/blob/9d60bc0a3ca4e201250254c866c4eedc1441ed4e/nav2_bringup/params/nav2_params.yaml
+# https://github.com/ros-navigation/navigation2/blob/3ee33f73a211f7e75c6f147644ec6ff275b94c88/nav2_bringup/params/nav2_params.yaml
 # Modification:
 #   - Set initial_pose for AMCL
 #   - Use stamped_cmd_vel
@@ -7,6 +7,7 @@
 /**:
   ros__parameters:
     enable_stamped_cmd_vel: true
+
 amcl:
   ros__parameters:
     alpha1: 0.2
@@ -118,6 +119,7 @@ controller_server:
       ax_max: 3.0
       ax_min: -3.0
       ay_max: 3.0
+      ay_min: -3.0
       az_max: 3.5
       vx_std: 0.2
       vy_std: 0.2
@@ -166,8 +168,9 @@ controller_server:
         enabled: true
         cost_power: 1
         cost_weight: 3.81
+        near_collision_cost: 253
         critical_cost: 300.0
-        consider_footprint: true
+        consider_footprint: false
         collision_cost: 1000000.0
         near_goal_distance: 1.0
         trajectory_point_step: 2
@@ -437,6 +440,13 @@ docking_server:
       k_delta: 2.0
       v_linear_min: 0.15
       v_linear_max: 0.15
+      use_collision_detection: true
+      costmap_topic: "local_costmap/costmap_raw"
+      footprint_topic: "local_costmap/published_footprint"
+      transform_tolerance: 0.1
+      projection_time: 5.0
+      simulation_step: 0.1
+      dock_collision_threshold: 0.3
 
 loopback_simulator:
   ros__parameters:
@@ -445,3 +455,9 @@ loopback_simulator:
     map_frame_id: "map"
     scan_frame_id: "base_scan"  # tb4_loopback_simulator.launch.py remaps to 'rplidar_link'
     update_duration: 0.02
+    scan_range_min: 0.05
+    scan_range_max: 30.0
+    scan_angle_min: -3.1415
+    scan_angle_max: 3.1415
+    scan_angle_increment: 0.02617
+    scan_use_inf: true

